tests/test_habkeuzes.py

```diff
@@ -0,0 +1,378 @@
+            kwaliteit=Kwaliteit.NVT,
+            idx_in_dt=None,
+            mits=GeenCriterium(),
+            mozaiek=GeenMozaiekregel(),
+            match_level=MatchLevel.NO_MATCH,
+        )
+
+    @classmethod
+    def HXXXX_niet_geautomatiseerd_SBB(cls, info: "VegTypeInfo"):
+        assert len(info.SBB) > 0
+        return cls(
+            onderbouwend_vegtype=info.SBB[0],
+            vegtype_in_dt=None,
+            habtype="HXXXX",
+            kwaliteit=Kwaliteit.NVT,
+            idx_in_dt=None,
+            mits=GeenCriterium(),
+            mozaiek=GeenMozaiekregel(),
+            match_level=MatchLevel.NO_MATCH,
+        )
+
+
+@dataclass
+class HabitatKeuze:
+    status: KeuzeStatus
+    habtype: str  # format = "H1123"
+    kwaliteit: Kwaliteit
+    habitatvoorstellen: List[HabitatVoorstel]  # used as a refence
+    opmerking: str = ""
+    mits_opmerking: str = ""
+    mozaiek_opmerking: str = ""
+    debug_info: Optional[str] = ""
+
+    def __post_init__(self):
+        # Validatie
+        if self.status in [
+            KeuzeStatus.DUIDELIJK,
+        ]:
+            assert self.habtype not in ["HXXXX", "H0000"]
+        elif self.status in [
+            KeuzeStatus.GEEN_KLOPPENDE_MITSEN,
+            KeuzeStatus.VEGTYPEN_NIET_IN_DEFTABEL,
+            KeuzeStatus.GEEN_OPGEGEVEN_VEGTYPEN,
+        ]:
+            assert self.habtype == "H0000"
+        elif self.status in [
+            KeuzeStatus.WACHTEN_OP_MOZAIEK,
+            KeuzeStatus.NIET_GEAUTOMATISEERD_CRITERIUM,
+            KeuzeStatus.MEERDERE_KLOPPENDE_MITSEN,
+        ]:
+            assert self.habtype == "HXXXX"
+
+        if self.habtype in ["H0000", "HXXXX"]:
+            assert self.kwaliteit == Kwaliteit.NVT
+
+    @classmethod
+    def habitatkeuze_for_postponed_mozaiekregel(
+        cls, habitatvoorstellen: List[HabitatVoorstel]
+    ):
+        return cls(
+            status=KeuzeStatus.WACHTEN_OP_MOZAIEK,
+            habtype="HXXXX",
+            kwaliteit=Kwaliteit.NVT,
+            opmerking="Er is een mozaiekregel waarvoor nog te weinig info is om een keuze te maken.",
+            habitatvoorstellen=habitatvoorstellen,
+            mits_opmerking="",
+            mozaiek_opmerking="",
+            debug_info="",
+        )
+
+    @property
+    def zelfstandig(self):
+        if self.habtype in ["H0000", "HXXXX"]:
+            return True
+
+        return is_mozaiek_type_present(self.habitatvoorstellen, GeenMozaiekregel)
+'''
+
+
+def test_habtype_toegekend():
+    voorstel = [
+        HabitatVoorstel(
+            onderbouwend_vegtype=SBB("25a1a"),
+            vegtype_in_dt=SBB("25a1a"),
+            habtype="H1234",
+            kwaliteit=Kwaliteit.NVT,
+            idx_in_dt=None,
+            mits=GeenCriterium(),
+            mozaiek=GeenMozaiekregel(),
+            match_level=MatchLevel.SUBASSOCIATIE_SBB,
+        )
+    ]
+    for v in voorstel:
+        v.mits.check(gpd.GeoSeries())
+        v.mozaiek.check(dict)
```

Comment:
Je geeft elke keer de dict constructor mee, dat zou niet moeten werken. Of `{}` of `dict()`

Reply:
Scherp. Het werkte omdat het GeenMozaiekregels zijn en die doen toch niks met de dict. Is gefixt nu.